src/device-registry/utils/generate-filter.js

```diff
@@ -549,15 +549,15 @@ const generateFilter = {
       }
 
       if (channel) {
-        filter["device_number"] = channel;
+        filter["device_number"] = parseInt(channel);
```

Comment:
resolving a runtime error around the query filtering for device_number. So casting string to Number.